lib/syntax_tree/node.rb

```diff
@@ -8325,6 +8347,27 @@ def ===(other)
         keyword_rest === other.keyword_rest && block === other.block
     end
 
+    # Returns a range representing the possible number of arguments accepted
+    # by this params node not including the block. For example:
+    #   def foo(a, b = 1, c:, d: 2, &block)
+    #     ...
+    #   end
+    # has arity 2..4
+    def arity
+      optional_keywords = keywords.count { |_label, value| value }
+      lower_bound =
+        requireds.length + posts.length + keywords.length - optional_keywords
+
+      upper_bound =
+        if keyword_rest || rest
+          Float::INFINITY
```

Comment:
I think `nil` here would be more canonical. `1..` will already allow you to do `.cover?` correctly.